moveit_simple/include/moveit_simple/online_robot.h

```diff
@@ -72,8 +72,53 @@ class OnlineRobot : public Robot
    * @throws <moveit_simple::CollisionDetected> (One of interpolated point is
    * in Collision with itself or environment)
    */
-  void execute(std::vector<moveit_simple::JointTrajectoryPoint> &goal,
-    bool collision_check = false);
+  void execute(std::vector<moveit_simple::JointTrajectoryPoint> &goal, bool collision_check = false);
+
+  /**
+   * @brief Set the goal to execute a given trajectory
+   * @param traj_name - name of trajectory to be executed (must be filled with
+   * prior calls to "addTrajPoint".
+   * @param collision_check - bool to turn check for collision on\off
+   */
+  bool setExecuteGoal(const std::string &traj_name, bool collision_check = false);
```

Comment:
What does the return mean?

Reply:
Can you think of an application where you would call `setExecuteGoal` and not immediately call `startExecution`?  Perhaps these two methods are always called together and should be combined into a `beginExecute` method?

Reply:
> What does the return mean?

The return is a boolean value, which in c++ is true or false. It means the goal was successfully set.

> Can you think of an application where you would call setExecuteGoal and not immediately call startExecution?

Yes.  In our case we calculate a timeout to cancel the execution in case it is stuck in isExecuting() but the trajectory isn't supposed to take that long.

Reply:
Do you need that functionality?  The current `execute` method does not currently allow for this.

Reply:
The current execute function has a timeout.

Reply:
Ok...but can you make the interface cleaner, less clunky.  Could the class take care of checking for timed out motion in some way?